Makefile

```diff
@@ -1,4 +1,17 @@
 # ---------------------------------------------------------------------
+# Build configuration
+# ---------------------------------------------------------------------
+
+BUILD_DIRECTORY ?= release
+
+ifndef BUILD_DIRECTORY
```

Comment:
Oh, I see now. I'll update the PR in a minute.

> Hmmm, interesting thought: BUILD_DIRECTORY=/tmp make clean will try to delete all of /tmp which probably isn't what we want ;-) The best fix for that might be to have make clean delete specifically-named files / dirs, rather than the whole $(BUILD_DIRECTORY).

Check my latest changes. I've updated `clean` to only remove things inside `$(BUILD_DIRECTORY)`, but not the directory itself.